packages/salesforcedx-vscode-core/test/vscode-integration/commands/functions/forceFunctionStart/ForceFunctionContainerStartExecutor.test.ts

```diff
@@ -0,0 +1,548 @@
+
+    const executor = new ForceFunctionContainerStartExecutor(
+      START_KEY,
+      LOG_NAME
+    );
+    // sets the functionsBinary
+    await executor.setupFunctionListeners('/does/not/matter', {
+      dispose: stub()
+    });
+    await executor.cancelFunction(disposable);
+    assert.calledOnce(fakeBinary.cancel);
+    assert.calledOnce(disposable.dispose);
+  });
+
+  describe('buildFunction', () => {
+    const functionName = 'imaFunction';
+    const functionDirPath = '/some/where/out/there';
+    it('Should throw an error if no binary is found.', async () => {
+      const executor = new ForceFunctionContainerStartExecutor(
+        START_KEY,
+        LOG_NAME
+      );
+
+      try {
+        await executor.buildFunction(functionName, functionDirPath);
+        fail('test should have failed due to no binary');
+      } catch (e) {
+        expect(e.message).contains(
+          'Unable to find binary for building function.'
+        );
+      }
+    });
+
+    it('Should be able to build the function.', async () => {
+      const fakeBinary = { build: stub(), on: stub() };
+      getFunctionsBinaryStub.resolves(fakeBinary);
+
+      const executor = new ForceFunctionContainerStartExecutor(
+        START_KEY,
+        LOG_NAME
+      );
+      // sets the functionsBinary
+      await executor.setupFunctionListeners('/does/not/matter', {
+        dispose: stub()
+      });
+      await executor.buildFunction(functionName, functionDirPath);
+      assert.calledOnce(appendLineStub);
+      assert.calledWith(fakeBinary.build, functionName, {
+        verbose: true,
+        path: functionDirPath
+      });
+    });
+  });
+
+  describe('startFunction', () => {
+    const functionName = 'imaFunctionToo';
+    const functionDirPath = '/neither/here/nor/there';
+    it('Should throw an error if not binary is found.', async () => {
+      const executor = new ForceFunctionContainerStartExecutor(
+        START_KEY,
+        LOG_NAME
+      );
+
+      try {
+        await executor.startFunction(functionName, functionDirPath);
+        fail('test should have failed to start due to no binary');
+      } catch (e) {
+        expect(e.message).contains('Unable to start function with no binary.');
+      }
+    });
+
+    it('Should be able to start a function.', async () => {
+      const fakeBinary = { run: stub(), on: stub() };
+      fakeBinary.run.resolves();
+      getFunctionsBinaryStub.resolves(fakeBinary);
+
+      const executor = new ForceFunctionContainerStartExecutor(
+        START_KEY,
+        LOG_NAME
+      );
+      // sets the functionsBinary
+      await executor.setupFunctionListeners('/does/not/matter', {
+        dispose: stub()
+      });
+      await executor.startFunction(functionName, functionDirPath);
+      assert.calledOnce(appendLineStub);
+      assert.calledWith(fakeBinary.run, functionName, {});
+    });
+  });
+
+  describe('ForceFunctionStartExecutor run.', () => {
```

Comment:
This exercises the method in the base class that is used by both the container and containerless executors.  I could move it to a standalone unit test and implement my own class extending it.  Lemme know your thoughts.